### ISSUE TITLE
CUDA inference produces NaN logits on GTX 1650 (fp16 instability) — works fine with CPU fallback

### ISSUE THREAD

**Issue by 94bash** [banjuman/vilm-lyrics-aligner#3]
When running alignment with Processing device: NVIDIA GPU · CUDA on a GTX 1650, the process fails during the Whisper transcription step with:

lyrics-aligner: Expected parameter logits (Tensor of shape (1, 51865)) of distribution Categorical(logits: torch.Size([1, 51865])) to satisfy the constraint IndependentConstraint(Real(), 1), but found invalid values:
tensor([[nan, nan, nan,  ..., nan, nan, nan]], device='cuda:0')
System.InvalidOperationException: lyrics-aligner: Expected parameter logits...
   at VilmLyricsAligner.Desktop.EngineBridge.RunAlignmentAsync(AlignmentRequest request, Action`1 onOutput, CancellationToken cancellationToken)
   at VilmLyricsAligner.Desktop.MainWindow.Generate_Click(Object sender, RoutedEventArgs e)

Steps to reproduce:

Load a ~170s audio file (WAV, 48kHz/24bit).
Paste full lyrics (Spanish).
Set Processing device to "NVIDIA GPU · CUDA" (GTX 1650, 4GB VRAM).
Run Auto mode alignment.
Fails during "Loading the alignment model and comparing the vocals…" with the NaN error above.

Workaround found: switching Processing device to CPU avoids the crash entirely. The log confirms why:

UserWarning: FP16 is not supported on CPU; using FP32 instead

This strongly suggests the issue is specifically fp16 numerical instability on this GPU, consistent with widely reported issues in the base OpenAI Whisper repo on GTX-series cards (which lack Tensor Cores / have weaker fp16 support than RTX cards), e.g. [openai/whisper#1068](https://github.com/openai/whisper/discussions/1068), [#383](https://github.com/openai/whisper/discussions/383).

Suggested fix: force fp32 (or detect known-problematic GPU architectures like GTX 16-series) even when CUDA is selected, or expose an "fp16/fp32" toggle in Settings so users with older/consumer GPUs aren't forced to fall back to full CPU mode (which is significantly slower) just to get a stable result.

Environment:

Windows 10/11, 64-bit
GPU: NVIDIA GTX 1650
Vilm Lyrics Aligner (installer version from voiceandfilm.com)
Audio: 170.10s WAV, 48kHz/24-bit
Lyrics: Spanish
